xilem_web/web_examples/streams/src/main.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This example demonstrates the use of [`memoized_stream`].
+
+use std::pin::Pin;
+
+use futures::{stream, Stream};
+use xilem_web::{
+    concurrent::memoized_stream, core::fork, document_body, elements::html,
+    input_event_target_value, interfaces::Element, App,
+};
+
+mod api;
+use self::api::{AbortHandle, MockConnection, StreamMessage};
+
+#[derive(Default)]
+struct AppState {
+    search_term: String,
+    db: MockConnection,
+    current_search: Option<AbortHandle>,
+}
+
+const DEBOUNCE_MILLIS: usize = 500;
+
+fn app_logic(state: &mut AppState) -> impl Element<AppState> {
+    log::debug!("Run app logic");
+
+    let search_stream = memoized_stream(
+        (state.search_term.clone(), state.db.clone()),
```

Comment:
This is a bit messy, because the first argument (`state.search_term`) is what should be monitored,
the second (`state.db`) is only there to make it available for stream generation (in `create_search_stream`).

Reply:
@Philipp-M 
I have now changed the implementation so that the app state is injected again before the future is initialized.
This has two advantages:
1. Only the data that is to be monitored for changes needs to implement `PartialEq` and  `Clone`
2. the point in time at which the future is created is now easier to grasp.

The API now looks like this:

```rust
memoized_await(
    50, // data to watch
    |state, watched_data| {
        state.async_call_started = true; // before it was debounced
        my_aync_call(state.non_watched_data, watched_data)
    },
    |state, result| { / * ... */ }
)

```